src/cgr_gwas_qc/workflow/modules/bcf.smk

```diff
@@ -0,0 +1,106 @@
+from cgr_gwas_qc import load_config
+from math import ceil
+
+cfg = load_config()
+
+############################
+# General BCF import rules
+###############################
+
+
+def _create_unknown_sex(wildcards):
+    path_to_unknown_sex_lst = temp("unknown_sex.lst")
+    cfg.ss.rename(columns={"Sample_ID": "#IID"}).assign(Sex="U").to_csv(
+        path_to_unknown_sex_lst, sep="\t", index=False
+    )
+    return path_to_unknown_sex_lst
+
+
+rule convert_bcf_to_plink_bed:
+    """Converts BCF to plink BED file
+
+    Path to aggregated BCF file is expected in user_files in config. The expected
+    BCF should be created using BCFtools/gtc2vcf. The BCF will be converted to BED
+    file for processing.
+    """
+    input:
+        bcf=cfg.config.user_files.bcf,
+    params:
+        prefix="sample_level/samples",
+        unknown_sex=_create_unknown_sex,
+    output:
+        bed="sample_level/samples.bed",
+        bim="sample_level/samples.bim",
+        fam="sample_level/samples.fam",
+    log:
+        "sample_level/samples.log",
+    conda:
+        cfg.conda("plink2-0")
+    benchmark:
+        "benchmarks/convert_bcf_to_plink_bed." + str(len(cfg.ss)) + ".tsv"
+    threads: 44
+    resources:
+        mem_mb=lambda wildcards, attempt: 1024 * 8 * attempt,
+        time_hr=ceil((0.11 * len(cfg.ss)) / 3600),
```

Comment:
I have revised in def5c7f3fe797cc1c6172da626843bfab86debf4. Now `mem_mb` will also scale with samples rather than multiple attempts.

I felt the same way when I was implementing it orignally (`mem_mb` with multiple attempts and scaling `time_hr` with samples). The reason was that the `mem_mb` usage is pretty low in this step. I have updated it anyway to scale with the number of samples. The rule should be completed in a single attempt and I would be curious to know a case when it fails due to time or memory requested.  If we run into scenerio that insufficient time or memory was requested, the scaling factor `0.11` should be increased appropriately.

I arrived at this equation by testing serially with multiple sample sizes and observing memory used and time taken.